src/Microsoft.ML.StandardLearners/Standard/Online/LinearSvm.cs

```diff
@@ -69,7 +69,8 @@ public sealed class Options : OnlineLinearOptions
             /// Column to use for example weight.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Column to use for example weight", ShortName = "weight", SortOrder = 4, Visibility = ArgumentAttribute.VisibilityType.EntryPointsOnly)]
-            public Optional<string> WeightColumn = Optional<string>.Implicit(DefaultColumnNames.Weight);
+            [BestFriend]
+            internal Optional<string> WeightColumn = Optional<string>.Implicit(DefaultColumnNames.Weight);
```

Comment:
>internal Optional<string> WeightColumn = Optional<string>.Implicit(DefaultColumnNames.Weight); [](start = 12, length = 94)

How user suppose to specify weight column during training?
This algorithm supports weights column in dataview see line 149

Reply:
@Ivanidzo4ka is correct... this must continue to be supported.

To remind and possibly rephrase earlier conversations since they were somewhat informal:

1. You will rename this field `WeightColumn` something appropriate, like, say, `WeightColumnOptional`. To maintain the entry-point name, the `Argument` attribute will have its `Name` property set to `WeightColumn`. (Thereby making the entry-points remain unchanged.)

2. You will introduce another field or property named `WeightColumn` intended for use from the API. This will be a public field, unadorned by an argument attribute, as it is not intended for API use. This will be a `string`, not an `Optional<string>`.

3. The code in LightGBM will appropriately adjust to the presence of either of these, thereby responding appropriately whether one or the other is set (or, as it may happen, not set). This may require some additional (hidden!) parameters somewhere to distinguish whether something was called from the command line/entry-points (in which case we'd expect the argument attribute field to be set), or the public surface of the API. There are many obvious ways of doing this, I'll trust we can figure out which is best.

The same pattern should be followed in other places where you are providing an "alternative" to the options. We want our `Options` object, where exposed, to expose all available options. (This in accordance of our usual attitude towards how these objects should behave, as captured in #1798.)

Reply:
As agreed made Weight and GroupId columns explicit strings

---
In reply to: [258773450](https://github.com/dotnet/machinelearning/pull/2674#discussion_r258773450) [](ancestors = 258773450)